build_scripts/windows/scripts/build.cmd

```diff
@@ -121,32 +121,42 @@ if not exist %PYTHON_DIR% (
     del get-pip.py
     echo Pip set up successful
 
-    dir .
+    REM setuptools is not installed by default in Python 3.12, but it is required by some dependencys
+    REM See https://github.com/Azure/azure-cli/pull/27196
+    REM Install wheel to force pip install azure-cli in legacy mode
+    REM see https://github.com/Azure/azure-cli/pull/29887
+    echo Installing setuptools wheel
+    %PYTHON_DIR%\python.exe -Im pip install setuptools wheel
```

Comment:
For `setuptools`:
Although `setuptools` is added to `azure-cli `dependency list in https://github.com/Azure/azure-cli/pull/27196, `azure-cli` is installed with `--no-deps`. Need to install it manually.

For `wheel`:
In 3.11, `wheel` is also installed when running `get-pip.py`. However, `wheel` is not included in final package in Linux and macOS. `setuptools` also removed `wheel` dependency in https://github.com/pypa/setuptools/issues/1386.

However, `wheel` is still a factor for `pip` to use pep517 or not: https://github.com/pypa/pip/blob/ec5faeac4ef6bae97df0e779566ceb2b0de89d3f/src/pip/_internal/pyproject.py#L107 (ref: https://github.com/pypa/packaging.python.org/issues/1517). If `wheel` is not installed, `pip` will use pep517 and fails, see https://github.com/Azure/azure-cli/pull/29887#issuecomment-2413201921. So I also install `wheel`. (I think it's okay to remove `wheel`)

```
# 3.11's get-pip.py

Successfully installed pip-24.2 setuptools-75.1.0 wheel-0.44.0
Pip set up successful
```